src/LayoutManager/LayoutManager.ts

```diff
@@ -55,7 +65,18 @@ export class LayoutManager {
     this.onBeforeLayout(strictContext);
 
     const layoutResult = this.getLayoutResult(strictContext);
-    layoutResult && this.commitLayout(strictContext, layoutResult);
+    if (layoutResult) {
+      this.commitLayout(strictContext, layoutResult);
+    } else {
+      // if there is no layout result, we may still need to flag the group as dirty
+      // if we are moving it through an active selection
+      if (
+        context.type === LAYOUT_TYPE_OBJECT_MODIFIED ||
+        context.type === LAYOUT_TYPE_OBJECT_MODIFYING
+      ) {
+        context.target.set('dirty', true);
+      }
+    }
```

Comment:
@ShaMan123 this is fix 2, for the group that weren't updating visually.

Reply:
I am not sure I understand why this is needed
What is the case in which there is no layout result?
The subscription will trigger the layout and it should have a result because the child transformed.

Reply:
not necessarly, indeed it doesn't have.
The clipPath strategy for example rarely does the layout.
And also the clipPath strategy does not need to redo a layout when you move an object around

Reply:
I still think it is not the LayoutManager's responsibility to do this part...

Reply:
and anyways it belongs to ActiveSelectionLayoutManager

Reply:
> The clipPath strategy for example rarely does the layout.
> And also the clipPath strategy does not need to redo a layout when you move an object around

Are you sure about this? An object moving changes the size, hence changes the center, hence changes the clip path position

I am not sure anymore.

IMO the correct approach is to delegate this responsibility to the ActiveSelection.
Consider the following. I have a multi nested active selection. I use `activeSelection.set` to change some transform prop (angle, scale).
The Layout API expects the dev to call performLayout at this point (a different discussion if it should trigger by itself).
If the dev doesn't call `triggerLayout` and groups are caching visuals will be wrong regardless of layout. Consider a object that was scaled now covers its sibling in the unchanged group area.

Reply:
Should this happen regardless of ActiveSelection?
In clip path layout?

You said that interactive group and layout are deeply coupled.
What about caching and everything?

Reply:
`What about caching and everything` does not make sense
@ShaMan123 is not worth.
I think is also better to just do blunt cache invalidation in the method always rather than even using the if since it will be needed anyway.

Reply:
That sounds better to me

Reply:
So then we should remove the if and just do it

Reply:
Or maybe even do it on `Group#onAfterLayout`